File: plugin/src/main/java/com/sun/plugin/MainActivity.java
package com.sun.plugin;

import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;
import com.qihoo360.replugin.loader.a.PluginActivity;

public class MainActivity extends PluginActivity {

    @Override
    protected void attachBaseContext(Context newBase) {
        super.attachBaseContext(newBase);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.plugin_main);
        TextView text = findViewById(R.id.text);
        text.setText(Build.MANUFACTURER);
        Log.i("MyPlugin", "classLoader:" + getClassLoader());
    }
}

File: app/src/main/java/com/sun/replugin/MainActivity.kt
package com.sun.replugin

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.qihoo360.replugin.RePlugin

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.start_plugin_btn).setOnClickListener {

            // 刻意以“包名”来打开
            RePlugin.startActivity(
                this@MainActivity,
                RePlugin.createIntent(
                    "com.sun.plugin",
                    "com.sun.plugin.MainActivity"
                )
            )
        }
    }
}